From 9eddb072da11f13c693de2963bdb2fb9606bb521 Mon Sep 17 00:00:00 2001
From: "Steven Rostedt (Red Hat)" <rostedt@goodmis.org>
Date: Mon, 12 Sep 2016 14:11:54 -0400
Subject: [PATCH 096/134] trace-cmd listen: Free pid_list in
 destroy_all_readers()

valgrind showed that pid_list created by create_all_readers() was never
freed. The obvious place to free it is in destroy_all_readers().

Signed-off-by: Steven Rostedt <rostedt@goodmis.org>
---
 trace-listen.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/trace-listen.c b/trace-listen.c
index a0d9945fffb1..71244cb40754 100644
--- a/trace-listen.c
+++ b/trace-listen.c
@@ -509,6 +509,8 @@ static void destroy_all_readers(int cpus, int *pid_array, const char *node,
 			pid_array[cpu] = 0;
 		}
 	}
+
+	free(pid_array);
 }
 
 static int *create_all_readers(int cpus, const char *node, const char *port,
-- 
2.5.5

